adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java

```diff
@@ -152,8 +167,27 @@ private void validateAcquireTokenRequest(final AuthenticationRequest authenticat
         }
 
         // validate authority
+        Telemetry.getInstance().startEvent(authenticationRequest.getTelemetryRequestId(),
+                EventStrings.AUTHORITY_VALIDATION_EVENT);
+        APIEvent apiEvent = new APIEvent(EventStrings.AUTHORITY_VALIDATION_EVENT);
```

Comment:
as you said in the other place, its going to be a big change. lets look at this later.

---
In reply to: [89234024](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/736#discussion_r89234024) [](ancestors = 89234024)